pkg/server/skydb/access.go

```diff
@@ -440,6 +446,35 @@ func (r FieldUserRole) Match(userinfo *UserInfo, record *Record) bool {
 	}
 }
 
+// matchDynamic is a helper function for returning whether the
+// field user role with dynamic user field type matches the specified
+// UserInfo and Record.
+func (r FieldUserRole) matchDynamic(userInfo *UserInfo, record *Record) bool {
+	dynamicFieldName := r.Data
+	switch fieldVal := record.Get(dynamicFieldName).(type) {
+	case string:
+		return userInfo.ID == fieldVal
+	case []interface{}:
+		for _, item := range fieldVal {
+			if userID, ok := item.(string); ok && userID == userInfo.ID {
+				return true
+			}
+		}
+		return false
+	}
+	return false
+}
+
+// Applicable returns true if the field user role is able to evaluate
+// the specified access mode.
+func (r FieldUserRole) Applicable(accessMode FieldAccessMode) bool {
+	if r.Type.RecordDependent() {
+		return accessMode == ReadFieldAccessMode ||
+			accessMode == WriteFieldAccessMode
+	}
```

Comment:
On second thought, `Match` functions already checked for cases that is covered by the `Applicable` function. This is because if we are doing a query, the `record` var is `nil`, and the `Match` function exists early returning `false`, which is exactly what `Applicable` will return.